src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -262,34 +266,31 @@ internal sealed class RSession : IRSession, IRCallbacks {
 
             await _initializationTcs.Task;
             initializationCts.Dispose();
+            _stopHostLock.EnqueueReset();
             lockToken.Set();
         }
 
-        public IRSessionSwitchBrokerTransaction StartSwitchingBroker() {
-            _disposeToken.ThrowIfDisposed();
-            return new BrokerTransaction(this);
-        }
+        public IRSessionSwitchBrokerTransaction StartSwitchingBroker() => !_disposeToken.IsDisposed ? new BrokerTransaction(this) : null;
 
-        public IRSessionReconnectTransaction StartReconnecting() {
-            _disposeToken.ThrowIfDisposed();
-            return new BrokerTransaction(this);
-        }
+        public IRSessionReconnectTransaction StartReconnecting() => !_disposeToken.IsDisposed ? new BrokerTransaction(this) : null;
 
         public async Task StopHostAsync() {
             _disposeToken.ThrowIfDisposed();
```

Comment:
Fixed

---
In reply to: [84113015](https://github.com/Microsoft/RTVS/pull/2529#discussion_r84113015) [](ancestors = 84113015,83911726)